### DISCUSSION TITLE
Allow "Switch Channels" window to search/switch across teams

### DISCUSSION THREAD

**Discussion by zefhemel** [mattermost/community#9] [Enhancements]
We all live in the Command-K/Ctrl-k channel switcher, but it only supports switching to channels in your current team. This has been the most frustrating aspects of using Mattermost with multiple teams since I joined the company — I cannot remember which channel is in what team, and... I don't care.

<img width="644" alt="CleanShot 2021-06-22 at 16 16 01" src="https://user-images.githubusercontent.com/59073/122940881-288e9780-d375-11eb-8144-d1826efd8cd0.png">

The idea is: let's make the channel switcher cross-team, or at least have a version that enables it.

**Comment by crspeller** [comment]
I actually worked on this for a hackathon. In progress PR here: https://github.com/mattermost/mattermost-webapp/pull/7217

**Comment by bbodenmiller** [comment]
Would be great if I could find channels across all teams even if I'm not yet part of that team or channel too (https://github.com/mattermost/community/discussions/118).

Update: not all channels, just channels I would have access to (e.g. public channels on a public team I haven't yet joined).

**Reply by Willyfrog** [reply]
I think there would be security concerns with that, I see that for you would be motivated by curiosity, but some other people might take advantage of it.

e.g.: suppose we had a super-secret-project-channel we are not yet ready to communicate.

**Reply by esethna** [reply]
Depends if the team is open or not though. If it's an "open" team that anyone can join then I don't see any security concerns with showing public channels in those teams. However this could clutter search results

**Reply by bbodenmiller** [reply]
> suppose we had a super-secret-project-channel we are not yet ready to communicate.

@Willyfrog sounds like it should be a private channel then. Security by obscurity is not secure.

**Reply by Willyfrog** [reply]
Given how our public channels work, if you are not in the team: it is essentially private for you.

If we had _really public_ channels in the sense that someone could read without registering, then those should appear. But if it requires registration and being in the team and you are not in that one, you shouldn't be able to browse.

This restriction could be controlled by a setting but that one should be turned off by default.

It is not a matter of security by obscurity (which I agree it's not the best), it is about not leaking data without knowing.

Don't get me wrong, I really like the idea of sharing and making as much stuff on the open as I can. But I also think that a user shouldn't share something in the public without being aware of it.

**Reply by Willyfrog** [reply]
> Depends if the team is open or not though. If it's an "open" team that anyone can join then I don't see any security concerns with showing public channels in those teams. However this could clutter search results

That is probably the closest we have to being really public, so yes that might be one were it might be acceptable.

**Reply by bbodenmiller** [reply]
@Willyfrog the issue I'm seeing is people creating redundant channels because they didn't know one already existed that they could access for their given topic.

**Reply by Willyfrog** [reply]
but if there exist a channel in a team you don't have access, it wouldn't solve that, right?

In general I'm ok with this as long as we take privacy into account and let the team admin decide on it :)

**Reply by chetanyakan** [reply]
In this case, should system admins see all public/private channels as they can join by links?

**Reply by esethna** [reply]
> but if there exist a channel in a team you don't have access, it wouldn't solve that, right?

Agree, we wouldn't want to show channels from other teams in this case.
